### PR TITLE
[SYCL][Driver] Fix spirv-ext passed to the translator by linker

### PR DESCRIPTION
**Pull request by MrSidims**
Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
@mdtoguchi @hchilama could you please advise, where should I add a test for this change?

**Comment by hchilama** [comment]
> @mdtoguchi @hchilama could you please advise, where should I add a test for this change?

Hi @MrSidims, may be you can use "clang/test/Driver/sycl-spirv-ext.c" to add your test for this change.

**Comment by MrSidims** [comment]
> > @mdtoguchi @hchilama could you please advise, where should I add a test for this change?
> 
> Hi @MrSidims, may be you can use "clang/test/Driver/sycl-spirv-ext.c" to add your test for this change.

Well, I've added sycl-spirv-ext.c for a bit of a different purpose - to check what extensions are passed to the translator in case if clang++ -fsycl is called. This should be a different test, for example: sycl-spirv-ext-linker.c and I was trying to find an already existing test to be used as a template. But then I realized that this particular lines aren't not only tested, but actually dead lines of code, see:
lib/Driver/ToolChains/SYCL.cpp
```
336       const char *LLVMSpirvOutputFile = constructLLVMSpirvCommand(
337           C, JA, Output, Prefix, true, II.getFilename());
```

so `ToBc` flag passed into `constructLLVMSpirvCommand` is always `true` (and the code above is the only use of this function). So the questions now are:
1. Do we want to remove the dead code or it's supposed to be used at some point of time? Tagging @mdtoguchi @hchilama @AGindinson 
2. If we want to leave these lines alone, then would it be acceptable to merge this patch without an appropriate test?

File: clang/lib/Driver/ToolChains/SYCL.cpp [hunk 1]
```diff
@@ -65,7 +65,46 @@ const char *SYCL::Linker::constructLLVMSpirvCommand(
...
+      // since it adds new storage classes that represent global_device and
+      // global_host address spaces, which are not supported for all
+      // targets. With the extension disabled the storage classes will be
+      // lowered to CrossWorkgroup storage class that is mapped to just
+      // global address space.
+      ExtArg += ",+SPV_INTEL_usm_storage_classes";
+    else
+      // Don't enable several freshly added extensions on FPGA H/W
+      ExtArg += ",+SPV_INTEL_token_type";
+    CmdArgs.push_back(ExtArg.data());
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/SYCL.cpp:107]: Having these settings in 2 spots can bring up potential problems where we miss a change from one spot to the other.  Is it possible to converge the usage to use the `SPIRVTranslatorJobAction`?

**Comment by MrSidims** [comment]
After https://github.com/intel/llvm/pull/5294 there is no longer need for this patch.

**Pull request closed without merging**